app/pages/ServiceListingPage/ServiceListingPage.tsx

```diff
@@ -41,10 +42,30 @@ export const ServiceListingPage = () => {
   useEffect(() => window.scrollTo(0, 0), []);
 
   useEffect(() => {
-    fetchService(id).then((s) => setService(s));
-    // TODO Handle Errors
+    fetchService(id).then((s) => {
+      if (typeof s === "string") {
+        setService(null);
+
+        setError(s);
+      } else {
+        setService(s);
+      }
```

Comment:
I don't love this because it means this implementation is tightly coupled to  the response result. Need to think more on it. 🤔